compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -1782,12 +1782,24 @@ static LogicalResult setRootDefaultConfig(IREE::GPU::TargetAttr target,
                                                preferredSubgroupSize);
 }
 
+/// Returns true if it's MatVec like i.e., either the bound of M or N dim = 1,
+/// or one of M, N dim isn't present.
 static bool isMatvecLike(linalg::LinalgOp linalgOp) {
-  if (linalgOp.getNumParallelLoops() != 2)
-    return false;
 
-  if (linalgOp.getNumReductionLoops() != 1)
+  auto bounds = linalgOp.getStaticLoopRanges();
+  SmallVector<unsigned> parallelDims;
+  linalgOp.getParallelDims(parallelDims);
+
+  // Validate that there's exactly one parallel dimension with size != 1.
+  auto nonUnitParallelDimsCount = llvm::count_if(
```

Comment:
This could also be `parallelDims.size() - count(parallelDims, 1)`, no?

Reply:
Will that make any difference?

Reply:
We'll have to check the bounds to obtain the size. parallelDims contain information about the indices.

Reply:
Ah sorry, I misread the code. Nevermind then.